plugins/nf-tower/src/main/io/seqera/tower/plugin/cli/AuthCommandImpl.groovy

```diff
@@ -0,0 +1,1354 @@
+package io.seqera.tower.plugin.cli
+
+import groovy.json.JsonBuilder
+import groovy.json.JsonSlurper
+import groovy.transform.Canonical
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import io.seqera.http.HxClient
+import nextflow.Const
+import nextflow.SysEnv
+import nextflow.cli.CmdAuth
+import nextflow.cli.ColorUtil
+import nextflow.config.ConfigBuilder
+import nextflow.exception.AbortOperationException
+
+import java.awt.Desktop
+import java.net.URI
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.StandardOpenOption
+import java.time.Duration
+
+@Slf4j
+@CompileStatic
+class AuthCommandImpl implements CmdAuth.AuthCommand {
+
+    static final int API_TIMEOUT_MS = 10000
+    static final int AUTH_POLL_TIMEOUT_RETRIES = 60
+    static final int AUTH_POLL_INTERVAL_SECONDS = 5
+    static final int WORKSPACE_SELECTION_THRESHOLD = 8  // Max workspaces to show in single list; above this uses org-first selection
+    private static final String DEFAULT_API_ENDPOINT = 'https://api.cloud.seqera.io'
+
+    private static final Map SEQERA_API_TO_AUTH0 = [
+        'https://api.cloud.dev-seqera.io'  : [
+            domain  : 'seqera-development.eu.auth0.com',
+            clientId: 'Ep2LhYiYmuV9hhz0dH6dbXVq0S7s7SWZ'
+        ],
+        'https://api.cloud.stage-seqera.io': [
+            domain  : 'seqera-stage.eu.auth0.com',
+            clientId: '60cPDjI6YhoTPjyMTIBjGtxatSUwWswB'
+        ],
+        'https://api.cloud.seqera.io' : [
+            domain  : 'seqera.eu.auth0.com',
+            clientId: 'FxCM8EJ76nNeHUDidSHkZfT8VtsrhHeL'
+        ]
+    ]
+
+    /**
+     * Creates an HxClient instance with optional authentication token
+     */
+    protected HxClient createHttpClient(String authToken = null) {
+        final builder = HxClient.newBuilder()
+            .connectTimeout(Duration.ofMillis(API_TIMEOUT_MS))
+
+        if (authToken) {
+            builder.bearerToken(authToken)
+        }
+
+        return builder.build()
+    }
+
+    @Override
+    void login(String apiUrl) {
+
+        // Check if TOWER_ACCESS_TOKEN environment variable is set
+        final envToken = SysEnv.get('TOWER_ACCESS_TOKEN')
```

Comment:
Better to use `Platform.getAccessToken` 

https://github.com/nextflow-io/nextflow/blob/main/modules/nextflow/src/main/groovy/nextflow/platform/PlatformHelper.groovy#L39-L44